### DISCUSSION TITLE
A fork of the stack that uses Ariakit instead of Radix UI

### DISCUSSION THREAD

**Discussion by gnapse** [epicweb-dev/epic-stack#654] [Show and tell]
Radix UI is great, but so is [Ariakit](https://ariakit.org/). I'm more familiar with the latter, and figured it'd be a fun way to get acquainted with the Epic Stack if I started doing the change in a fork that would then be useful to start any project off of it.

Check it out at https://github.com/gnapse/epic-stack

**Comment by rajeshg** [comment]
@gnapse My understanding is that @kentcdodds is interested in swapping radix ui with Draft UI (it's react aria based too). Not sure how different ariakit is from Draft UI. 

![image](https://github.com/epicweb-dev/epic-stack/assets/143390/a6b49f36-96f2-4b8b-a3c4-175a37b18412)

**Reply by kentcdodds** [reply]
I'm also considering https://www.jollyui.dev (which is also based on react-aria).

We've got good choices here and Radix is fine. But eventually I'll spend a little time on this and really re-evaluate them again.

**Reply by gnapse** [reply]
That's cool. I agree that it should be a user's choice. But also some default should be present. Perhaps the default in the official stack should be the library that is the simplest to remove and replace by the users adopting the stack. Plus, **the stack's skeleton app should rely on the UI library of choice as little as possible** (so that it is easier to remove and switch to a different one).

For instance, I do not see a reason for the skeleton stack app so use Radix-UI's `Label` or the checkbox components. It could've worked really well with the corresponding native HTML elements. Same with `Tooltip`. It can perfectly work with `title` attributes. Users adopting the stack will know if they want a fancy tooltip and incorporate it themselves.

Anyway, I created this fork precisely because I wanted to use Ariakit instead of Radix UI. It was not overly complex to do the change. But it could've been easier if the app would rely less on Radix UI from the start.

**Reply by kentcdodds** [reply]
Adaptability is a goal, but so is demonstrating proper usage and limiting analysis paralysis. A title is not the same as a tooltip from an accessibility standpoint. I don't want people to have to do a lot of work if they're fine with the decided opinions (which most people should be). The difference between Ariakit and Radix are mostly insubstantial in the grand scheme of things anyway.

The epic stack targets folks who just want to get stuff done rather than work on wiring things together and making a bunch of decisions and I think it does a good job of that. It's also quite helpful for folks who have different opinions and want to use different tools, but that's not the target audience.

**Reply by gnapse** [reply]
Yes, that's fair. I knew that tooltips are better than title attributes, so I should've known better before offering that as an example.

I do stand by the checkbox and label, though. I'm unsure what the Radix ones offer that the native one does not. There surely is something, but I'm unsure if it is something that the use of those components is actually leveraging in the stack skeleton.

Anyway, is not a big deal. I love the stack as it is.

**Comment by kentcdodds** [comment]
That's great @gnapse! Forks are welcome! If you'd like to add it to the https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md then I'd happily merge :)

**Reply by gnapse** [reply]
Great idea. Done! https://github.com/epicweb-dev/epic-stack/pull/656
